ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -53,48 +57,53 @@ export const DataTree = ({
   </Box>
 );
 
-const DataField = ({
-  label,
-  isPermit,
-  type,
-  value,
-  tokenDecimals,
-}: {
-  label: string;
-  isPermit: boolean;
-  type: string;
-  value: ValueType;
-  tokenDecimals: number;
-}) => {
-  if (typeof value === 'object' && value !== null) {
-    return (
-      <DataTree
-        data={value}
-        isPermit={isPermit}
-        tokenDecimals={tokenDecimals}
-      />
-    );
-  }
-  if (isPermit && label === 'value') {
-    return (
-      <ConfirmInfoRowText
-        text={formatNumber(
-          parseInt(value, 10) / Math.pow(10, tokenDecimals),
-          tokenDecimals,
-        )}
-      />
-    );
-  }
-  if (isPermit && label === 'deadline') {
-    return <ConfirmInfoRowDate date={parseInt(value, 10)} />;
-  }
-  if (
-    type === 'address' &&
-    isValidHexAddress(value, {
-      mixedCaseUseChecksum: true,
-    })
-  ) {
-    return <ConfirmInfoRowAddress address={value} />;
-  }
-  return <ConfirmInfoRowText text={sanitizeString(value)} />;
-};
+const DataField = memo(
+  ({
+    label,
+    isPermit,
+    type,
+    value,
+    tokenDecimals,
+  }: {
+    label: string;
+    isPermit: boolean;
+    type: string;
+    value: ValueType;
+    tokenDecimals: number;
+  }) => {
+    if (typeof value === 'object' && value !== null) {
+      return (
+        <DataTree
+          data={value}
+          isPermit={isPermit}
+          tokenDecimals={tokenDecimals}
+        />
+      );
+    }
+    if (isPermit && label === 'value') {
+      const valueBN = new BigNumber(
```

Comment:
Minor, but could we split this function up with some local components such as `DataFieldNumber`?

Reply:
There is more requirement for formatting I will do some refactoring in DataTree in upcoming PRs.